src/common/idle_inhibitor.cpp

```diff
@@ -0,0 +1,231 @@
+                                   dbus_uint32_t* serial);
+    DBusMessage* (*connection_send_with_reply_and_block)(DBusConnection* connection,
+                                                         DBusMessage* message,
+                                                         int timeout_milliseconds,
+                                                         DBusError* error);
+    dbus_bool_t (*message_get_args)(DBusMessage* message, DBusError* error, int first_arg_type,
+                                    ...);
+    void (*message_unref)(DBusMessage* message);
+
+    bool initialized = false;
+    void* handle = nullptr;
+
+    DBus() {
+        handle = dlopen("libdbus-1.so.3", RTLD_NOW);
+        if (handle) {
+#define dbus_symbol(sym)                                                                           \
+    ((sym = reinterpret_cast<decltype(sym)>(dlsym(handle, "dbus_" #sym))), sym != nullptr)
+            initialized = dbus_symbol(bus_get);
+            initialized |= dbus_symbol(connection_unref);
+            initialized |= dbus_symbol(message_new_method_call);
+            initialized |= dbus_symbol(message_append_args);
+            initialized |= dbus_symbol(connection_send);
+            initialized |= dbus_symbol(connection_send_with_reply_and_block);
+            initialized |= dbus_symbol(message_get_args);
+            initialized |= dbus_symbol(message_unref);
+#undef dbus_symbol
+            if (!initialized) {
+                LOG_ERROR(Common, "Failed to load libdbus functions");
+            }
+        } else {
+            LOG_WARNING(Common, "Failed to dlopen() libdbus-1.so.3");
+        }
+    }
+
+    ~DBus() {
+        if (handle)
+            dlclose(handle);
+    }
+};
+
+DBus g_dbus;
+#endif
+} // namespace
+
+namespace Common {
+IdleInhibitor::IdleInhibitor() {
+#if defined(__unix__)
+    ConnectDBus();
+#endif
+}
+
+IdleInhibitor::~IdleInhibitor() {
+    if (inhibiting_idle)
+        AllowIdle();
+#ifdef __unix__
+    if (dbus_connection)
+        g_dbus.connection_unref(dbus_connection);
+#endif
+}
+
+#if defined(_WIN32)
+void IdleInhibitor::InhibitIdle() {
+    inhibiting_idle = true;
+    SetThreadExecutionState(ES_CONTINUOUS | ES_SYSTEM_REQUIRED | ES_DISPLAY_REQUIRED);
+}
+
+void AllowIdle() {
+    inhibiting_idle = false;
+    SetThreadExecutionState(ES_CONTINUOUS);
+}
+
+#elif defined(__unix__)
+void IdleInhibitor::ResetScreensaver() {
+    auto poll_time = std::chrono::steady_clock::now() + std::chrono::seconds(10);
+    for (;;) {
+        std::unique_lock<std::mutex> lk(cond_mutex);
+        poll_cond.wait_until(lk, poll_time, [this]() { return !inhibiting_idle; });
+        if (!inhibiting_idle)
+            return;
+
+        if (fork() == 0)
+            execlp("xdg-screensaver", "xdg-screensaver", "reset", nullptr);
+
+        poll_time += std::chrono::seconds(10);
+    }
+}
+
+void IdleInhibitor::ConnectDBus() {
+    if (g_dbus.initialized) {
```

Comment:
Inverting this into a guard clause allows the code to be unindented a level.

```suggestion
    if (!g_dbus.initialized) {
        return;
    }
```